File: internal/model/invite.go
package model

import "time"

type InviteRelationshipStatus string

const (
	InviteRelationshipStatusPending  InviteRelationshipStatus = "pending"
	InviteRelationshipStatusRewarded InviteRelationshipStatus = "rewarded"
)

type InviteRewardEventStatus string

const (
	InviteRewardEventStatusGranted  InviteRewardEventStatus = "granted"
	InviteRewardEventStatusReversed InviteRewardEventStatus = "reversed"
)

type InviteRewardBeneficiaryRole string

const (
	InviteRewardBeneficiaryInviter InviteRewardBeneficiaryRole = "inviter"
	InviteRewardBeneficiaryInvitee InviteRewardBeneficiaryRole = "invitee"
)

type InviteRelationship struct {
	ID                     string                   `json:"id"`
	InviterUserID          string                   `json:"inviterUserId"`
	InviterCode            string                   `json:"inviterCode"`
	InviteeUserID          string                   `json:"inviteeUserId"`
	Status                 InviteRelationshipStatus `json:"status"`
	FirstPaidOrderID       string                   `json:"firstPaidOrderId"`
	FirstPaidOrderNo       string                   `json:"firstPaidOrderNo"`
	FirstPaidOrderKind     string                   `json:"firstPaidOrderKind"`
	FirstPaidAmountCNYCent int64                    `json:"firstPaidAmountCnyCent"`
	FirstPaidAt            *time.Time               `json:"firstPaidAt"`
	RewardedAt             *time.Time               `json:"rewardedAt"`
	LastReversedAt         *time.Time               `json:"lastReversedAt"`
	CreatedAt              time.Time                `json:"createdAt"`
	UpdatedAt              time.Time                `json:"updatedAt"`
}

type InviteRewardEvent struct {
	ID                     string                      `json:"id"`
	RelationID             string                      `json:"relationId"`
	BeneficiaryUserID      string                      `json:"beneficiaryUserId"`
	BeneficiaryRole        InviteRewardBeneficiaryRole `json:"beneficiaryRole"`
	OrderID                string                      `json:"orderId"`
	OrderNo                string                      `json:"orderNo"`
	Status                 InviteRewardEventStatus     `json:"status"`
	AmountMicros           int64                       `json:"amountMicros"`
	OrderPaidAmountCNYCent int64                       `json:"orderPaidAmountCnyCent"`
	CreatedAt              time.Time                   `json:"createdAt"`
}

type InviteConfigResponse struct {
	Enabled             bool  `json:"enabled"`
	ConfigComplete      bool  `json:"configComplete"`
	InviterRewardMicros int64 `json:"inviterRewardMicros"`
	InviteeRewardMicros int64 `json:"inviteeRewardMicros"`
	MinFirstPaidCNYCent int64 `json:"minFirstPaidCnyCent"`
}

type InviteConfigRequest struct {
	Enabled             bool  `json:"enabled"`
	InviterRewardMicros int64 `json:"inviterRewardMicros"`
	InviteeRewardMicros int64 `json:"inviteeRewardMicros"`
	MinFirstPaidCNYCent int64 `json:"minFirstPaidCnyCent"`
}

type InviteSummaryResponse struct {
	Enabled           bool   `json:"enabled"`
	ConfigComplete    bool   `json:"configComplete"`
	InviteCode        string `json:"inviteCode"`
	InvitedUsers      int64  `json:"invitedUsers"`
	PendingInvites    int64  `json:"pendingInvites"`
	RewardedInvites   int64  `json:"rewardedInvites"`
	TotalRewardMicros int64  `json:"totalRewardMicros"`
	InvitedByUserID   string `json:"invitedByUserId"`
	InvitedByUsername string `json:"invitedByUsername"`
	BoundInviteCode   string `json:"boundInviteCode"`
}

type InviteRewardEventResponse struct {
	ID                     string                      `json:"id"`
	RelationID             string                      `json:"relationId"`
	BeneficiaryUserID      string                      `json:"beneficiaryUserId"`
	BeneficiaryUsername    string                      `json:"beneficiaryUsername"`
	BeneficiaryRole        InviteRewardBeneficiaryRole `json:"beneficiaryRole"`
	OrderNo                string                      `json:"orderNo"`
	Status                 InviteRewardEventStatus     `json:"status"`
	AmountMicros           int64                       `json:"amountMicros"`
	OrderPaidAmountCNYCent int64                       `json:"orderPaidAmountCnyCent"`
	CreatedAt              time.Time                   `json:"createdAt"`
}

type InviteRelationshipAdminResponse struct {
	ID                     string                   `json:"id"`
	InviterUserID          string                   `json:"inviterUserId"`
	InviterUsername        string                   `json:"inviterUsername"`
	InviterCode            string                   `json:"inviterCode"`
	InviteeUserID          string                   `json:"inviteeUserId"`
	InviteeUsername        string                   `json:"inviteeUsername"`
	Status                 InviteRelationshipStatus `json:"status"`
	FirstPaidOrderNo       string                   `json:"firstPaidOrderNo"`
	FirstPaidOrderKind     string                   `json:"firstPaidOrderKind"`
	FirstPaidAmountCNYCent int64                    `json:"firstPaidAmountCnyCent"`
	FirstPaidAt            *time.Time               `json:"firstPaidAt"`
	RewardedAt             *time.Time               `json:"rewardedAt"`
	LastReversedAt         *time.Time               `json:"lastReversedAt"`
	CreatedAt              time.Time                `json:"createdAt"`
}

type InviteStatsResponse struct {
	Enabled              bool  `json:"enabled"`
	ConfigComplete       bool  `json:"configComplete"`
	RelationsTotal       int64 `json:"relationsTotal"`
	PendingTotal         int64 `json:"pendingTotal"`
	RewardedTotal        int64 `json:"rewardedTotal"`
	GrantedRewardMicros  int64 `json:"grantedRewardMicros"`
	ReversedRewardMicros int64 `json:"reversedRewardMicros"`
}

File: web/src/api/invite.ts
import { authFetch } from './client'

const API_BASE = '/api'

async function fetchJson<T>(url: string, options?: RequestInit): Promise<T> {
  const res = await authFetch(url, options)
  let data: unknown
  try {
    data = await res.json()
  } catch {
    data = undefined
  }
  if (!res.ok) {
    const payload = data as { error?: string } | undefined
    throw new Error(payload?.error || `请求失败 (${res.status})`)
  }
  return data as T
}

export interface InviteConfig {
  enabled: boolean
  configComplete: boolean
  inviterRewardMicros: number
  inviteeRewardMicros: number
  minFirstPaidCnyCent: number
}

export interface InviteSummary {
  enabled: boolean
  configComplete: boolean
  inviteCode: string
  invitedUsers: number
  pendingInvites: number
  rewardedInvites: number
  totalRewardMicros: number
  invitedByUserId: string
  invitedByUsername: string
  boundInviteCode: string
}

export interface InviteRewardEvent {
  id: string
  relationId: string
  beneficiaryUserId: string
  beneficiaryUsername: string
  beneficiaryRole: 'inviter' | 'invitee'
  orderNo: string
  status: 'granted' | 'reversed'
  amountMicros: number
  orderPaidAmountCnyCent: number
  createdAt: string
}

export interface InviteStats {
  enabled: boolean
  configComplete: boolean
  relationsTotal: number
  pendingTotal: number
  rewardedTotal: number
  grantedRewardMicros: number
  reversedRewardMicros: number
}

export interface InviteRelation {
  id: string
  inviterUserId: string
  inviterUsername: string
  inviterCode: string
  inviteeUserId: string
  inviteeUsername: string
  status: 'pending' | 'rewarded'
  firstPaidOrderNo: string
  firstPaidOrderKind: string
  firstPaidAmountCnyCent: number
  firstPaidAt?: string | null
  rewardedAt?: string | null
  lastReversedAt?: string | null
  createdAt: string
}

export async function getMyInviteSummary(): Promise<InviteSummary> {
  return fetchJson<InviteSummary>(`${API_BASE}/me/invite/summary`)
}

export async function listMyInviteRewards(): Promise<InviteRewardEvent[]> {
  const data = await fetchJson<{ items: InviteRewardEvent[] }>(`${API_BASE}/me/invite/rewards`)
  return data.items || []
}

export async function getInviteConfig(): Promise<InviteConfig> {
  return fetchJson<InviteConfig>(`${API_BASE}/admin/invite/config`)
}

export async function updateInviteConfig(payload: Omit<InviteConfig, 'configComplete'>): Promise<InviteConfig> {
  const data = await fetchJson<{ config: InviteConfig }>(`${API_BASE}/admin/invite/config`, {
    method: 'PUT',
    body: JSON.stringify(payload),
  })
  return data.config
}

export async function getInviteStats(): Promise<InviteStats> {
  return fetchJson<InviteStats>(`${API_BASE}/admin/invite/stats`)
}

export async function listInviteRelations(): Promise<InviteRelation[]> {
  const data = await fetchJson<{ items: InviteRelation[] }>(`${API_BASE}/admin/invite/relations`)
  return data.items || []
}

export async function listInviteRewardEvents(): Promise<InviteRewardEvent[]> {
  const data = await fetchJson<{ items: InviteRewardEvent[] }>(`${API_BASE}/admin/invite/reward-events`)
  return data.items || []
}
